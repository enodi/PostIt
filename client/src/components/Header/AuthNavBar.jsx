import React from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { signoutUser } from '../../actions/auth/signinAction';

/**
 * This class is the container component
 * for the different NavBar
 * It is responsible for managing state
 * change and handling user signout
 *
 * @class AuthNavBar
 * @extends {React.Component}
 */
class AuthNavBar extends React.Component {

  /**
   * Creates an instance of AuthNavBar.
   * @param {object} props
   *
   * @memberof AuthNavBar
   */
  constructor(props) {
    super(props);

    this.handleOnClick = this.handleOnClick.bind(this);
  }

  /**
   * Handles user signout and removing token
   * from localStorage
   *
   * @param {object} event
   *
   * @returns {void}
   *
   * @memberof AuthNavBar
   */
  handleOnClick(event) {
    event.preventDefault();
    this.props.signoutUser();
  }

  /**
   *
   *
   * @returns {void} jsx - xml/html like syntax extension to javascript
   *
   * @memberof AuthNavBar
   */
  render() {
    return (
      <nav className="white dashboard logout-icon">
      <div className="nav-wrapper">
        <Link to="/" className="brand-logo li">PostIt</Link>
        <ul className="right hide-on-med-and-down">
          <li>
            <i
              className="large material-icons black-text"
              onClick={this.handleOnClick}>input
            </i>
          </li>
        </ul>
        <ul className="side-nav" id="mobile-demo">
          <li>
            <i className="large material-icons black-text">
              input
            </i>
          </li>
        </ul>
      </div>
    </nav>
    );
  }
}

export default connect(null, { signoutUser })(AuthNavBar);
