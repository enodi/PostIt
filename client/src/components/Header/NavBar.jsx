import React from 'react';
import { Link, IndexLink } from 'react-router';

/**
 * NavBar Presentational Component
 *
 * @param {object} props
 *
 * @returns {jsx} an xml/html like syntax extension for
 * javascript
 */
const NavBar = () => (
  <nav>
    <div className="nav-wrapper #212121 grey darken-4">
      <IndexLink to="/" className="brand-logo">
        <span
          className="logo grey-text text-lighten-2"
        >PostIt
        </span>
      </IndexLink>
      <Link to="#"
        data-activates="mobile-demo"
        className="button-collapse">
        <i className="material-icons">menu</i>
      </Link>
      <ul className="right hide-on-med-and-down">
        <li><a href="http://postit-enodi.herokuapp.com/apidocs/#introduction">
        API Docs</a></li>
        <li><Link to="/account">My Account</Link></li>
      </ul>
      <ul className="side-nav" id="mobile-demo">
        <li><Link to="/account">My Account</Link></li>
      </ul>
    </div>
  </nav>
);

export default NavBar;
