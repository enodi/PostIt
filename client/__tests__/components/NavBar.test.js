import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../../src/components/Header/NavBar.jsx';
import AuthNavBar from '../../src/components/Header/AuthNavBar.jsx';

describe('NavBar Component', () => {
  it('should render NavBar Component', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('AuthNavBar Component', () => {
  it('should render AuthNavBar Component', () => {
    const wrapper = shallow(<AuthNavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});

