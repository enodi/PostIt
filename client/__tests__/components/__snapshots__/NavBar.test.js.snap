// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthNavBar Component should render AuthNavBar Component 1`] = `
<div>
  <nav
    className="white dashboard logout-icon"
  >
    <div
      className="nav-wrapper"
    >
      <Link
        className="brand-logo li"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/"
      >
        PostIt
      </Link>
      <ul
        className="right hide-on-med-and-down"
      >
        <li>
          <a
            className="modal-trigger"
            href="#add_users"
          >
            <i
              className="material-icons black-text"
              href="#add_users"
            >
              group_add
            </i>
          </a>
        </li>
      </ul>
      <ul
        className="side-nav"
        id="mobile-demo"
      >
        <li>
          <i
            className="large material-icons black-text"
          >
            group_add
          </i>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`NavBar Component should render NavBar Component 1`] = `
<nav>
  <div
    className="nav-wrapper #212121 grey darken-4"
  >
    <IndexLink
      className="brand-logo"
      to="/"
    >
      <span
        className="logo grey-text text-lighten-2"
      >
        PostIt
      </span>
    </IndexLink>
    <Link
      className="button-collapse"
      data-activates="mobile-demo"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="#"
    >
      <i
        className="material-icons"
      >
        menu
      </i>
    </Link>
    <ul
      className="right hide-on-med-and-down"
    >
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/account"
        >
          My Account
        </Link>
      </li>
    </ul>
    <ul
      className="side-nav"
      id="mobile-demo"
    >
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/account"
        >
          My Account
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
